### ISSUE TITLE
Distributed limiters aren't communicating correctly

### ISSUE THREAD

**Issue by simonbyrne** [CliMA/ClimaCore.jl#1511]
## Describe the bug
The `QuasimonotoneLimiter` constructor calls `create_ghost_buffer`:
https://github.com/CliMA/ClimaCore.jl/blob/9c220472af5dab549275c183975a5b1fe735d0ba/src/Limiters/quasimonotone.jl#L53-L54

However in the only test, this currently returns `nothing`
https://github.com/CliMA/ClimaCore.jl/pull/1510

Probably not worth fixing at the moment: in #1494, we will probably change the limiters so that they will be applied only within the element (i.e. no communication should be required).

**Comment by charleskawczynski** [comment]
Yeah, it looks like `create_ghost_buffer` needs to support `VIFH` data layouts.

**Comment by simonbyrne** [comment]
it will also throw an error later, since we don't actually define any methods for `GhostBuffer` (I think I removed them all a while ago)
